packages/nuxt/src/pages/utils.ts

```diff
@@ -215,114 +218,135 @@ export function extractScriptContent (sfc: string) {
+    for (const macro of script.code.matchAll(PAGE_EXTRACT_RE)) {
+      found[macro[1]!] = false
+    }
+
+    if (Object.keys(found).length === 0) {
       continue
     }
 
     const dynamicProperties = new Set<keyof NuxtPage>()
 
-    let foundMeta = false
-
     parseAndWalk(script.code, absolutePath.replace(/\.\w+$/, '.' + script.loader), (node) => {
-      if (foundMeta) { return }
+      if (node.type !== 'ExpressionStatement' || node.expression.type !== 'CallExpression' || node.expression.callee.type !== 'Identifier') { return }
 
-      if (node.type !== 'ExpressionStatement' || node.expression.type !== 'CallExpression' || node.expression.callee.type !== 'Identifier' || node.expression.callee.name !== 'definePageMeta') { return }
+      // function name is one of the extracted macro functions and not yet found
+      const fnName = node.expression.callee.name
+      if (fnName in found === false || found[fnName] !== false) { return }
+      found[fnName] = true
 
-      foundMeta = true
       let code = script.code
-      let pageMetaArgument = node.expression.arguments[0]
-      if (pageMetaArgument?.type !== 'ObjectExpression') {
-        logger.warn(`\`definePageMeta\` must be called with an object literal (reading \`${absolutePath}\`).`)
-        return
-      }
+      let pageExtractArgument = node.expression.arguments[0]
 
-      // when using ts we slice, transform and parse the `definePageMeta` node to avoid parsing the whole file
+      // TODO: always true because `extractScriptContent` only detects ts/tsx loader
       if (/tsx?/.test(script.loader)) {
+        // slice, transform and parse the `define...` macro node to avoid parsing the whole file
         const transformed = oxcTransform(absolutePath, script.code.slice(node.start, node.end), { lang: script.loader })
-        code = transformed.code
         if (transformed.errors.length) {
           for (const error of transformed.errors) {
-            logger.warn('Error while transforming `definePageMeta()`' + error.codeframe)
+            logger.warn(`Error while transforming \`${fnName}()\`` + error.codeframe)
           }
           return
         }
 
         // we already know that the first statement is a call expression
-        pageMetaArgument = ((parseSync('', transformed.code, { lang: 'js' }).program.body[0]! as ExpressionStatement).expression as CallExpression).arguments[0]! as ObjectExpression
+        pageExtractArgument = ((parseSync('', transformed.code, { lang: 'js' }).program.body[0]! as ExpressionStatement).expression as CallExpression).arguments[0]
+        code = transformed.code
       }
 
-      for (const key of extractionKeys) {
-        const property = pageMetaArgument.properties.find((property): property is ObjectProperty => property.type === 'Property' && property.key.type === 'Identifier' && property.key.name === key)
-        if (!property) { continue }
+      if (pageExtractArgument?.type !== 'ObjectExpression') {
```

Comment:
Moving this check after transformation resolves an unreported bug/edge-case, when the argument expression itself has typescript assertions such as `definePageMeta({ path: '/' } as Record<string, unknown>)` (not as likely to run into as using typescript on properties though)